src/main/java/com/faforever/api/data/util/ElidePageSizeUtil.java

```diff
@@ -0,0 +1,58 @@
+package com.faforever.api.data.util;
+
+import com.faforever.api.error.ApiException;
+import com.faforever.api.error.ErrorCode;
+import com.yahoo.elide.core.security.User;
+import lombok.experimental.UtilityClass;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.util.MultiValueMap;
+
+import java.util.Optional;
+
+import static com.faforever.api.data.domain.GroupPermission.ROLE_SCRAPER;
+
+@UtilityClass
+public class ElidePageSizeUtil {
+
+  static final String PAGE_SIZE_PARAM = "page[size]";
+  static final String PAGE_LIMIT_PARAM = "page[limit]";
+
+
+  public void validatePageSize(final MultiValueMap<String, String> allRequestParams,
+    final User principal,
+    final int elideDefaultPageSize,
+    final int elideMaxPageSize) {
+
+    final int rolePageSizeLimit = getRolePageSizeLimit(principal, elideDefaultPageSize,
+      elideMaxPageSize);
+    final int pageSizeInRequest = getParamsPageSize(allRequestParams).orElse(elideDefaultPageSize);
+
+    if (pageSizeInRequest > rolePageSizeLimit) {
+      throw ApiException.of(ErrorCode.QUERY_INVALID_PAGE_SIZE, pageSizeInRequest,
+        rolePageSizeLimit);
+    }
```

Comment:
I would not throw an error here. Instead just go for min(requested or default, allowed)